src/shared/orgUtils.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+
+export class OrgUtils {
+  /**
+   * Given an app name, it queries the org to find the matching app id. To do so,
+   * it will first attempt at finding the app with a matching DeveloperName. If
+   * no match is found, it will then attempt at finding the app with a matching
+   * Label. If multiple matches are found, then the first match is returned.
+   *
+   * @param connection the connection to the org
+   * @param appName the name of the app
+   * @returns the app id or undefined if no match is found
+   */
+  public static async getAppId(connection: Connection, appName: string): Promise<string | undefined> {
+    // NOTE: We have to break up the query and run against different columns separately instead
+    // of using OR statement, otherwise we'll get the error 'Disjunctions not supported'
+    const devNameQuery = `SELECT DurableId FROM AppDefinition WHERE DeveloperName LIKE '${appName}'`;
```

Comment:
Is `LIKE` overly permissive for these queries? Should we not use `=`?

Reply:
No it is not. `LIKE` allows us to check while ignoring cases. It also can be used for pattern matching `but only if` we add pattern wildcards such as `%` Since here I am not adding any wildcards then `LIKE` will strictly match the name but ignoring the casing.